sdk/go/common/resource/plugin/provider_plugin.go

```diff
@@ -78,8 +79,12 @@ func NewProvider(host Host, ctx *Context, pkg tokens.Package, version *semver.Ve
 		})
 	}
 
-	plug, err := newPlugin(ctx, ctx.Pwd, path, fmt.Sprintf("%v (resource)", pkg),
-		[]string{host.ServerAddr()}, nil /*env*/)
+	args := []string{host.ServerAddr()}
+	for k, v := range options {
+		args = append(args, fmt.Sprintf("-%s=%v", k, v))
+	}
```

Comment:
To make this work for dynamic providers in Python, I needed some way to pass the `virtualenv` runtime option to the `pulumi-resource-pulumi-python` provider plugin. This change passes along all runtime options as command line flags to providers. Are we OK doing this? (I looked and all _our_ providers should be able to handle the additional command line flags; they'll be ignored). Should we prefix these flag names with `runtime-` or something else to make it more clear where these are coming from?

Reply:
Environment variables seem like a safer option here--have you given those a try?

Reply:
Environment variables should be able to work for this. I can update the PR to pass them as `PULUMI_RUNTIME_<OPTION>`.

Note that this does mean the provider plugins would be receiving these values differently than the language and analyzer plugins, which receive them as command line flags.